src/Container/FilterUsingXForwardedHeadersFactory.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mezzio\Container;
+
+use Laminas\Diactoros\ServerRequestFilter\FilterUsingXForwardedHeaders;
+use Mezzio\ConfigProvider;
+use Psr\Container\ContainerInterface;
+
+use function array_key_exists;
+use function is_array;
+use function is_string;
+
+/**
+ * Factory for use in generating a custom FilterUsingXForwardedHeaders instance.
+ *
+ * Assign this factory to the * Laminas\Diactoros\ServerRequestFilter\FilterServerRequestInterface service.
+ * Then define configuration as follows:
+ *
+ * <code>
+ * 'laminas-diactoros' => [
+ *     'server-request-filter' => [
+ *         'x-forwarded-headers' => [
+ *             // Trust any proxy:
+ *             'trusted-proxies' => '*',
+ *             // Trust specific proxies:
+ *             'trusted-proxies' => ['192.168.0.1', '192.168.0.2'],
+ *             // Trust entire subnets:
+ *             'trusted-proxies' => ['192.168.0.0/24', '10.0.0.0/16'],
+ *             // Trust specific X-Forwared headers:
+ *             'trusted-headers' => ['X-Forwarded-Host', 'X-Forwarded-Proto'],
+ *         ],
+ *     ],
+ * ],
+ * </code>
+*/
+final class FilterUsingXForwardedHeadersFactory
+{
+    public function __invoke(ContainerInterface $container): FilterUsingXForwardedHeaders
+    {
+        // phpcs:disable Generic.Files.LineLength.TooLong
+        $config = $container->get('config');
+        $config = $config[ConfigProvider::DIACTOROS_CONFIG_KEY][ConfigProvider::DIACTOROS_SERVER_REQUEST_FILTER_CONFIG_KEY][ConfigProvider::DIACTOROS_X_FORWARDED_FILTER_CONFIG_KEY] ?? [];
+
+        if (! is_array($config) || empty($config)) {
```

Comment:
I don't think we need to check `empty()`

Reply:
Psalm was complaining, IIRC. :)

Reply:
Actually... the question here is: what should we do if there is no trusted-proxies and/or trusted-headers configuration?

The assumption I made is that if they registered this factory, and then did not provide configuration, it would be as strict as possible — hence the check for `empty($config)`. Thoughts?

Reply:
If there are no such keys, don't we default to empty arrays anyway?

Reply:
It should raise an exception; it's invalid configuration. Change pushed.